error.go

```diff
@@ -100,17 +100,32 @@ func (d *ErrorDetail) Value() (proto.Message, error) {
 //
 // [the documentation on errors]: https://connect.build/docs/go/errors
 type Error struct {
-	code    Code
-	err     error
-	details []*ErrorDetail
-	meta    http.Header
+	code        Code
+	err         error
+	details     []*ErrorDetail
+	meta        http.Header
+	isServerErr bool
 }
 
 // NewError annotates any Go error with a status code.
 func NewError(c Code, underlying error) *Error {
 	return &Error{code: c, err: underlying}
 }
 
+// IsServerErr returns true if the error was produced by the server.
```

Comment:
Let's rethink this name a little; `IsServerErr` suggests to me that we're categorizing errors into 4xx and 5xx. Perhaps `IsFromWire`?

Also, can we give some examples in the GoDoc? I think the idea is approachable, but a little subtle.

Reply:
Although talking API design we probably want it to be `IsWireErr` because `connect.IsFromWire` doesn't give enough context